Update O2 defaults

dberzano opened the pull request:
* Bump ROOT 6 to v6-08-06-80-ge2951b4
* Allow optional builds of AliRoot using O2 defaults

dberzano commented:
@sawenzel this also implements the addition of `GCC-Toolchain` as a dependency, but note that - in order for that to be truly effective - we also need to update `root.sh` (another pr is coming, but we cannot merge it to master until we release the next AliRoot as it will cause a rebuild).

Can you check if this is OK, and really supersedes #745?

dberzano commented:
Also note that I deliberately did not touch `defaults-o2-daq.sh`.

dberzano commented:
As agreed with @sawenzel next time we will work on a single PR using the "Allow edits from maintainers" option (I have never used it).

sawenzel commented on defaults-o2-dev-fairroot.sh:
```diff
@@ -29,14 +27,25 @@ overrides:
       - Python-modules
+      - GCC-Toolchain:(?!osx)
+  AliRoot:
+    requires:
```
I am wondering: why do we need to list these things? Isn't this listed in the AliRoot recipe?

dberzano replied:
AliRoot dependencies for O2 must include ZeroMQ, which is not listed in the default `aliroot.sh` recipe. Does this answer your question?

sawenzel replied:
Partially. Why do we need ZeroMQ for AliRoot?

dberzano replied:
That is a good question that we should ask @mkrzewic maybe. Note that I would separate the ZeroMQ dependency problem from this PR, which just synchronizes `defaults-o2.sh` (that always had ZeroMQ) and `defaults-o2-dev-fairroot.sh`. If this is controversial I'll refactor the PR by removing this synchronization, no problem with me.

mkrzewic replied:
Hi, the HLT part of aliroot needs zeromq, also the event display part that speaks to the HLT for online display.  In general it makes sense to have this anywhere we have the zmq library installed - which is the case for O2 installs.

dberzano commented:
Note that I have:

* removed the addition of ZeroMQ to AliRoot (let's do it in a different PR if required)
* added the (accidentally) removed AliEn-Runtime dependency to ROOT

Basically, ROOT has to be built _with_ AliEn-Runtime... even if you will not build AliRoot.

I am not sure if this is exactly desirable... maybe, once again, two different defaults are really what we need?

pzhristov commented:
Since for the moment we do not have any replacement for AliEn, it is needed and has to be build. Lets proceed with the merge.

sawenzel commented:
What is blocking this PR to be merged?

sawenzel commented:
Is there any news on this? Without this PR (or related fixed) ROOT is still not compiled with the correct gcc compiler and AliRoot is still strictly disabled in o2-defaults.

ktf commented:
@dberzano do you have time to converge on this?

matthiasrichter commented on defaults-o2-dev-fairroot.sh:
```diff
@@ -5,8 +5,6 @@ env:
   CMAKE_BUILD_TYPE: "RELWITHDEBINFO"
 disable:
-  - AliEn-Runtime
-  - AliRoot
```
@dberzano why would you remove this, effectively enable the packages? Probably wrong for this default configuration.

sawenzel replied:
No no ... removing this line just does not prohibit building AliRoot with this default. It does not make AliRoot a dependency (at least with #817).

ktf commented:
Is this still relevant? Can we close it if not?

sawenzel commented:
I think it is still relevant. It allows building AliROOT with o2 defaults and correctly makes COMPILTER_TOOLCHAIN a prerequisite for ROOT.

sawenzel commented:
@ktf: This is the PR I was talking about (which correctly makes GCC_TOOLCHAIN a requirement to build ROOT).

sawenzel commented:
This PR can be closed I believe.
